otherlibs/unix/unix.mli

```diff
@@ -675,11 +675,39 @@ val close_process_full :
 (** {6 Symbolic links} *)
 
 
-val symlink : string -> string -> unit
-(** [symlink source dest] creates the file [dest] as a symbolic link
-   to the file [source].
-
-   On Windows: not implemented. *)
+val symlink : ?is_dir:bool -> string -> string -> unit
+(** [symlink ?is_dir source dest] creates the file [dest] as a symbolic link
+   to the file [source]. On Windows, [~is_dir] indicates if the symbolic link
+   points to a directory or a file; if omitted, [symlink] examines [source]
+   using [stat] and picks appropriately, if [source] does not exist then [false]
+   is assumed. On Unix, [~is_dir] ignored.
```

Comment:
I'd be very against option 2 and uneasy with option 3.

The problem with option 1 is that it *requires* you to open the target as part of creating the link. The target could be an (unavailable) network resource which introduces a delay - part of the purpose of specifying `~is_dir` is that at that point one simply accesses the underlying `CreateSymbolicLink` function. It feels weaker to have no ability to do that.